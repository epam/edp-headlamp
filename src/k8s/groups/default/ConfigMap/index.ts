import { K8s } from '@kinvolk/headlamp-plugin/lib';
import { streamResults } from '../../../common/streamResults';
import { ConfigMapKubeObjectConfig } from './config';
import { StreamListProps } from './types';

const {
  name: { pluralForm },
  version,
} = ConfigMapKubeObjectConfig;

export class ConfigMapKubeObject extends K8s.configMap.default {
  static streamList({ namespace, dataHandler, errorHandler }: StreamListProps): () => void {
    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}`;
    return streamResults(url, dataHandler, errorHandler);
  }
}
