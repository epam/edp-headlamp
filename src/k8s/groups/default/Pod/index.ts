import { K8s } from '@kinvolk/headlamp-plugin/lib';

export class PodKubeObject extends K8s.pod.default {}
