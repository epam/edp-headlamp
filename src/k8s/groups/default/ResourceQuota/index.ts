import { K8s } from '@kinvolk/headlamp-plugin/lib';
import { streamResults } from '../../../common/streamResults';
import { ResourceQuotaKubeObjectConfig } from './config';
import { RESOURCE_QUOTA_LABEL_TENANT } from './labels';
import { StreamListProps } from './types';

const {
  name: { pluralForm },
  version,
} = ResourceQuotaKubeObjectConfig;

export class ResourceQuotaKubeObject extends K8s.ResourceClasses.ResourceQuota {
  static streamList({
    namespace,
    tenantNamespace,
    dataHandler,
    errorHandler,
  }: StreamListProps): () => void {

    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}`;
    return streamResults(url, dataHandler, errorHandler, {
      labelSelector: `${RESOURCE_QUOTA_LABEL_TENANT}=edp-workload-${tenantNamespace}`,
    });
  }
}
