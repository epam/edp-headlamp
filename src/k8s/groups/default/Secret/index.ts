import { ApiProxy, K8s } from '@kinvolk/headlamp-plugin/lib';
import { STATUS_COLOR } from '../../../../constants/colors';
import { ICONS } from '../../../../icons/iconify-icons-mapping';
import { KubeObjectListInterface } from '../../../../types/k8s';
import { SecretKubeObjectConfig } from './config';
import { SECRET_LABEL_SECRET_TYPE } from './labels';
import { SecretKubeObjectInterface } from './types';

const {
  name: { pluralForm },
  version,
} = SecretKubeObjectConfig;

export class SecretKubeObject extends K8s.secret.default {
  static getClusterSecretList(
    namespace: string
  ): Promise<KubeObjectListInterface<SecretKubeObjectInterface>> {
    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}?labelSelector=${SECRET_LABEL_SECRET_TYPE}=cluster`;

    return ApiProxy.request(url);
  }

  static getSecretByName(namespace: string, name: string): Promise<SecretKubeObjectInterface> {
    const url = `/api/${version}/namespaces/${namespace}/${pluralForm}/${name}`;

    return ApiProxy.request(url);
  }

  static getStatusIcon(connected: string): [string, string, boolean?] {
    if (connected === undefined) {
      return [ICONS.UNKNOWN, STATUS_COLOR.UNKNOWN];
    }

    if (connected === 'true') {
      return [ICONS.CHECK_CIRCLE, STATUS_COLOR.SUCCESS];
    }
    return [ICONS.CROSS_CIRCLE, STATUS_COLOR.ERROR];
  }
}
