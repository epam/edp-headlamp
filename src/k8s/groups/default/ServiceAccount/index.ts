import { K8s } from '@kinvolk/headlamp-plugin/lib';

export class ServiceAccountKubeObject extends K8s.serviceAccount.default {}
