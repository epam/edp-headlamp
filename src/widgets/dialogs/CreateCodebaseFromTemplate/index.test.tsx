/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../../mocks/wrappers/default';
import { CreateCodebaseFromTemplateDialog } from '.';

test('renders CreateCodebaseFromTemplate Create component', () => {
  render(
    <TestWrapper>
      <CreateCodebaseFromTemplateDialog
        props={{
          template: {
            apiVersion: 'v2.edp.epam.com/v1alpha1',
            kind: 'Template',
            metadata: {
              name: 'antora',
              namespace: 'edp-delivery-vp-dev',
              creationTimestamp: '',
              uid: '',
            },
            spec: {
              buildTool: 'npm',
              category: 'docs',
              description:
                'Utilize the AsciiDoc markup language to construct, arrange, and manage your documentation while adopting a Documentation as Code methodology.',
              displayName: 'Documentation as Code with Antora',
              framework: 'antora',
              icon: [
                {
                  base64data:
                    'PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxZW0iIGhlaWdodD0iMWVtIiB2aWV3Qm94PSIwIDAgMzIgMzIiPjxnIGZpbGw9Im5vbmUiPjxwYXRoIGZpbGw9IiNGM0FENjEiIGQ9Ik0yOS4zNzQgMmgtNC4xN2EyMi40IDIyLjQgMCAwIDAtNi4xOTQuODc0bC0xLjk5IDMuMTdMMTQuMDEyIDVhMjIuMzk1IDIyLjM5NSAwIDAgMC0zLjI2OSAyLjNhMTkuMDI1IDE5LjAyNSAwIDAgMC02LjcyIDE0LjUxdjMuMjVzLjE3LS4zLjQ5LS44MWMtLjA5LjM4LS4xMy41OS0uMTMuNTl2LjIyaDMuOTJjMi4wOCAwIDQuMDgtLjQ3IDUuODktMS4zNGMuMjEtLjEuMjEtLjQgMC0uNWwtMi4xNy0xLjAxYy0uMTMtLjA0LS4wOS0uMjEuMDQtLjIxaDQuNTNjLjIxIDAgLjQxLS4wOC41Ny0uMjFjMS4zMy0xLjE0IDIuNDUtMi41NCAzLjI3LTQuMTVjLjMtLjYuODctMS42NyAxLjE4LTIuMjVjLjEzLS4yNC4xMi0uNTMtLjAyLS43N2wtMS4zNi0yLjI0YS4yNS4yNSAwIDAgMSAuMjEtLjM4bDEuNTMzLS45OTRsMi4wNDQtMS45OThsMi43Ni0yLjAwMmE0NS44NjEgNDUuODYxIDAgMCAxIDMuMDQzLTMuODE2Yy4xMy0uMTQuMi0uMzMuMi0uNTJhLjY1My42NTMgMCAwIDAtLjY1LS42N1oiLz48cGF0aCBmaWxsPSIjQTU2OTUzIiBkPSJNMTQgMTJoOS4xMTRjLjI2IDAgLjUxLS4xNC42NC0uMzZjLjk0LTEuNTcgMS45Ni0zLjE0IDMuMDQtNC42NGgtNy43OVYyLjg4QTIyLjEyOSAyMi4xMjkgMCAwIDAgMTQgNS4wMTZWMTJaIi8+PHBhdGggZmlsbD0iI0QzODgzRSIgZD0iTTI1LjI4NCA0LjE0YS40OTguNDk4IDAgMCAwLS42NS0uMjNjLTQuNiAyLjA3LTguNzQgNC45NC0xMi4zNCA4LjUzYTQxLjU4NCA0MS41ODQgMCAwIDAtOC45MiAxMy4yMmE0MS40NyA0MS40NyAwIDAgMC0xLjM1IDMuNjdjLS4wOS4zLjA4LjYxLjM4LjY2Yy4yNi4wNS41LS4xMS41Ny0uMzVjLjM4LTEuMjEuODItMi40MSAxLjMyLTMuNTljMi4wNC00LjgzIDQuOTctOS4xOCA4LjctMTIuOWMzLjUxLTMuNTEgNy41NS02LjMgMTIuMDQtOC4zMmEuNS41IDAgMCAwIC4yNS0uNjljMCAuMDEgMCAuMDEgMCAwWiIvPjwvZz48L3N2Zz4=',
                  mediatype: 'image/svg+xml',
                },
              ],
              keywords: ['antora', 'asciidoc', 'documentation', 'technical writing'],
              language: 'javascript',
              maintainers: [
                {
                  email: 'email',
                  name: 'name',
                },
              ],
              maturity: 'stable',
              minEDPVersion: '3.4.0',
              source: 'https://github.com/test',
              type: 'application',
              version: '0.1.0',
            },
          },
        }}
        state={{
          open: true,
          openDialog: jest.fn(),
          closeDialog: jest.fn(),
        }}
      />
    </TestWrapper>
  );

  const dialog = screen.getByTestId('dialog');
  expect(dialog).toMatchSnapshot();
});
