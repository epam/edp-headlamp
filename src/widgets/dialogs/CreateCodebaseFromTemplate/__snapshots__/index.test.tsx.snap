// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CreateCodebaseFromTemplate Create component 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="dialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby="mui-1"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthMd MuiDialog-paperFullWidth css-1ntld2w-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-8yphvn-MuiTypography-root-MuiDialogTitle-root"
        id="mui-1"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1pf3k2c-MuiTypography-root"
        >
          Create application from template
        </p>
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <div
          class="MuiBox-root css-178yklu"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-14dv046-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap css-x9mjaw-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <img
                        alt=""
                        src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxZW0iIGhlaWdodD0iMWVtIiB2aWV3Qm94PSIwIDAgMzIgMzIiPjxnIGZpbGw9Im5vbmUiPjxwYXRoIGZpbGw9IiNGM0FENjEiIGQ9Ik0yOS4zNzQgMmgtNC4xN2EyMi40IDIyLjQgMCAwIDAtNi4xOTQuODc0bC0xLjk5IDMuMTdMMTQuMDEyIDVhMjIuMzk1IDIyLjM5NSAwIDAgMC0zLjI2OSAyLjNhMTkuMDI1IDE5LjAyNSAwIDAgMC02LjcyIDE0LjUxdjMuMjVzLjE3LS4zLjQ5LS44MWMtLjA5LjM4LS4xMy41OS0uMTMuNTl2LjIyaDMuOTJjMi4wOCAwIDQuMDgtLjQ3IDUuODktMS4zNGMuMjEtLjEuMjEtLjQgMC0uNWwtMi4xNy0xLjAxYy0uMTMtLjA0LS4wOS0uMjEuMDQtLjIxaDQuNTNjLjIxIDAgLjQxLS4wOC41Ny0uMjFjMS4zMy0xLjE0IDIuNDUtMi41NCAzLjI3LTQuMTVjLjMtLjYuODctMS42NyAxLjE4LTIuMjVjLjEzLS4yNC4xMi0uNTMtLjAyLS43N2wtMS4zNi0yLjI0YS4yNS4yNSAwIDAgMSAuMjEtLjM4bDEuNTMzLS45OTRsMi4wNDQtMS45OThsMi43Ni0yLjAwMmE0NS44NjEgNDUuODYxIDAgMCAxIDMuMDQzLTMuODE2Yy4xMy0uMTQuMi0uMzMuMi0uNTJhLjY1My42NTMgMCAwIDAtLjY1LS42N1oiLz48cGF0aCBmaWxsPSIjQTU2OTUzIiBkPSJNMTQgMTJoOS4xMTRjLjI2IDAgLjUxLS4xNC42NC0uMzZjLjk0LTEuNTcgMS45Ni0zLjE0IDMuMDQtNC42NGgtNy43OVYyLjg4QTIyLjEyOSAyMi4xMjkgMCAwIDAgMTQgNS4wMTZWMTJaIi8+PHBhdGggZmlsbD0iI0QzODgzRSIgZD0iTTI1LjI4NCA0LjE0YS40OTguNDk4IDAgMCAwLS42NS0uMjNjLTQuNiAyLjA3LTguNzQgNC45NC0xMi4zNCA4LjUzYTQxLjU4NCA0MS41ODQgMCAwIDAtOC45MiAxMy4yMmE0MS40NyA0MS40NyAwIDAgMC0xLjM1IDMuNjdjLS4wOS4zLjA4LjYxLjM4LjY2Yy4yNi4wNS41LS4xMS41Ny0uMzVjLjM4LTEuMjEuODItMi40MSAxLjMyLTMuNTljMi4wNC00LjgzIDQuOTctOS4xOCA4LjctMTIuOWMzLjUxLTMuNTEgNy41NS02LjMgMTIuMDQtOC4zMmEuNS41IDAgMCAwIC4yNS0uNjljMCAuMDEgMCAuMDEgMCAwWiIvPjwvZz48L3N2Zz4="
                        style="width: 2.5rem;"
                      />
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <h5
                        aria-label="Documentation as Code with Antora"
                        class="MuiTypography-root MuiTypography-h5 makeStyles-templateName-2 css-xqgeko-MuiTypography-root"
                        data-mui-internal-clone-element="true"
                      >
                        Documentation as Code with Antora
                      </h5>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-ab4e19-MuiTypography-root"
              >
                Utilize the AsciiDoc markup language to construct, arrange, and manage your documentation while adopting a Documentation as Code methodology.
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-lndw90-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    Language
                  </p>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-7ldjcq-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <span
                        style="display: block; line-height: 0;"
                      >
                        <svg
                          height="20"
                          width="20"
                        >
                          <use
                            xlink:href="#javascript"
                          />
                        </svg>
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      JavaScript
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    Framework
                  </p>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-7ldjcq-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <span
                        style="display: block; line-height: 0;"
                      >
                        <svg
                          height="20"
                          width="20"
                        >
                          <use
                            xlink:href="#antora"
                          />
                        </svg>
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      Antora
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    Build Tool
                  </p>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-7ldjcq-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      <span
                        style="display: block; line-height: 0;"
                      >
                        <svg
                          height="20"
                          width="20"
                        >
                          <use
                            xlink:href="#npm"
                          />
                        </svg>
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                    >
                      NPM
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    Type
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-ab4e19-MuiTypography-root"
                  >
                    application
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    Category
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-ab4e19-MuiTypography-root"
                  >
                    docs
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    Maturity
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-ab4e19-MuiTypography-root"
                  >
                    stable
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    Version
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-ab4e19-MuiTypography-root"
                  >
                    0.1.0
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    Source
                  </p>
                  <a
                    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-9d13eh-MuiTypography-root-MuiLink-root"
                    href="https://github.com/test"
                    target="_blank"
                  >
                    https://github.com/test
                  </a>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    Maintainers
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    <div>
                      <span
                        class="MuiTypography-root MuiTypography-body2 css-ab4e19-MuiTypography-root"
                      >
                        Name:
                         
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1yjsbnr-MuiTypography-root"
                      >
                        name
                      </span>
                    </div>
                    <div>
                      <span
                        class="MuiTypography-root MuiTypography-body2 css-ab4e19-MuiTypography-root"
                      >
                        Email:
                         
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1yjsbnr-MuiTypography-root"
                      >
                        email
                      </span>
                    </div>
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-9vf8gf-MuiTypography-root"
                  >
                    Keywords
                  </p>
                  <span
                    class="MuiTypography-root MuiTypography-caption css-1yjsbnr-MuiTypography-root"
                  >
                    antora
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 css-1ezega9-MuiTypography-root"
                  >
                    , 
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-caption css-1yjsbnr-MuiTypography-root"
                  >
                    asciidoc
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 css-1ezega9-MuiTypography-root"
                  >
                    , 
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-caption css-1yjsbnr-MuiTypography-root"
                  >
                    documentation
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 css-1ezega9-MuiTypography-root"
                  >
                    , 
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-caption css-1yjsbnr-MuiTypography-root"
                  >
                    technical writing
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
      >
        <div
          class="MuiStack-root css-ee32gy-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorInherit css-17jf5nl-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            cancel
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-vw8gqz-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            proceed
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
